lasagne/tests/layers/test_input.py

```diff
@@ -31,3 +31,8 @@ def test_bad_shape_fails(self):
 
         with pytest.raises(ValueError):
             InputLayer((3, 2), input_var)
+
+    def test_negative_shape_fails(self):
+        from lasagne.layers.input import InputLayer
+        with pytest.raises(ValueError):
+            InputLayer((3, -2))
```

Comment:
Should check for zeros as well, not just for negative shapes. This test can basically be copy-pasted from `TestLayer.test_nonpositive_input_dims_raises_value_error()` in `test_base.py`.